File: src-old/cucumber/java/io/tripled/social/client/CucumberDateTimeProvider.java
package io.tripled.social.client;

import io.tripled.social.client.domain.DateTimeProvider;

public class CucumberDateTimeProvider implements DateTimeProvider {

  private Clock clock;

  public CucumberDateTimeProvider(Clock clock) {
    this.clock = clock;
  }

  @Override
  public LocalDateTime now() {
    return LocalDateTime.now(clock);
  }

  public void fixate(Instant instant) {
    clock = Clock.fixed(instant, ZoneId.systemDefault());
  }

  public void fixateWithOffset(Duration duration) {
    clock = Clock.offset(clock, duration);
  }
}

File: src/test/kotlin/io/tripled/acceptance/CucumberDateTimeProvider.kt
package io.tripled.acceptance

import io.tripled.social.client.domain.DateTimeProvider
import java.time.*

class CucumberDateTimeProvider internal constructor(private var clock: Clock) : DateTimeProvider {
    override fun now(): LocalDateTime {
        return LocalDateTime.now(clock)
    }

    fun fixate(instant: Instant) {
        clock = Clock.fixed(instant, ZoneId.systemDefault())
    }

    fun fixateWithOffset(duration: Duration) {
        clock = Clock.offset(clock, duration)
    }
}
